src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -325,7 +333,8 @@ private static ILogger CreateLogger(bool verbose, ReportFormat reportFormat, IRe
                     .AddSimpleConsole(options =>
                     {
                         options.SingleLine = true;
-                    });
+                    })
+                    .AddProvider(new SummaryLoggerProvider(loggedErrors, loggedWarnings));
```

Comment:
Instead of creating the dictionaries here and passing them into the provider to pass into the logger, can you create them in the summary logger, make them public properties, and get a reference to the summary logger and refer to them that way?  Seems best to have the logger own them instead of the CLI.